sys/arm64/arm64/machdep.c

```diff
@@ -567,6 +568,15 @@ try_load_dtb(caddr_t kmdp)
 	vm_pointer_t dtbp;
 
 	dtbp = MD_FETCH(kmdp, MODINFOMD_DTBP, vm_offset_t);
+#ifdef __CHERI_PURE_CAPABILITY__
+	if (dtbp != (vm_pointer_t)NULL) {
+		dtbp = (vm_pointer_t)cheri_andperm(cheri_setaddress(
+		    kernel_root_cap, dtbp), CHERI_PERMS_KERNEL_DATA);
+		dtbp = (vm_pointer_t)cheri_setbounds((void *)dtbp,
```

Comment:
You can probably drop the casts here.  Maybe the old code you copied from was from before cheri_* could operate on uintcap_t?

Reply:
Done, and dropped from riscv too